### PR TITLE
[KOGITO-8671] Update native env handling

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-8671

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [ ] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by desmax74 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -70,15 +73,6 @@ Table of Contents
         - [Running Bats tests](#running-bats-tests)
         - [Writing Bats tests](#writing-bats-tests)
     - [Reporting new issues](#reporting-new-issues)
-            [Image Modules](#image-modules)
```
Comment (spolti) [inline README.md:73]: why removing it?

File: README.md [hunk 2]
```diff
@@ -566,7 +560,7 @@ For this example, let's use the same as the previous one (process-quarkus-exampl
 But this time, let's perform a native build:
 
 ```bash
-$ mvn clean package -Pnative
+$ mvn clean package -Dnative
```
Comment (spolti) [inline README.md:563]: in the process-quarkus-example examples there is no native profile anymore nor something that can be activated with property. How does it work?

Comment (radtriste) [reply README.md:563]: parent pom: https://github.com/kiegroup/kogito-examples/blob/main/kogito-quarkus-examples/pom.xml#L104

File: README.md (hunk 1 above)
Comment (radtriste) [reply README.md:73]: Because this has nothing to do there and is already present a bit before into the TOC ...
My IDE automatically reformatted it :)

**Review by spolti — APPROVED** [review]

**Pull request merged**